translator.py

```diff
@@ -236,59 +255,83 @@ def _alloc_mcast_addr(self, client_addr):
-            if resp_get.status_code != 200:
-                errmsg += f'GET resulted in status code {resp_get.status_code}. Response body:\n{resp_get.text}'
-                print(errmsg)
-                return
-            csrftoken = sess.cookies.get('csrftoken')
-            if csrftoken is None:
-                errmsg += 'no csrftoken found in session.'
-                print(errmsg)
-                return
-            # We can now submit the form using a POST request. In doing so, we must provide the CSRF protection token
-            # alongside the other form parameters (the form has a hidden input tag that contains the CSRF protection
-            # token) and as a cookie (but the requests module takes care of this part automatically because we're using
-            # a session). Also note that we must set the Referer header field (to the same URL we're submitting the form
-            # to) as otherwise we'll get rejected by the server.
-            form_params = {'csrfmiddlewaretoken': csrftoken, 'source': self._mcast_src_ip, 'group': str(mcast_dst_ip),
-                           'udp_port': str(self._mcast_dst_port), 'email': email, 'description': str(description),
-                           'Add': 'Add'}
-            header_fields = {'Referer': constants.MULTICASTMENU_ADD_URL}
-            resp_post = sess.post(constants.MULTICASTMENU_ADD_URL, data=form_params, headers=header_fields)
-            if resp_post.status_code != 200:
+            options = {
+                "unique_identifier": self._mm_uid,
+                "source": self._mcast_src_ip,
+                "group": str(mcast_dst_ip),
+            }
+
+            if src_ip[0] == constants.MULTICASTMENU_SENDER_IP:
+                print("'Inside Request' set to True")
+                options["inside_request"] = "1"
+
+            resp_post = requests.post(constants.MULTICASTMENU_ADD_URL, data=options)
+
+            if resp_post.status_code != 201:
                 errmsg += f'POST resulted in status code {resp_post.status_code}. Response body:\n{resp_post.text}'
                 print(errmsg)
                 return
+
+            access_code = json.loads(resp_post.text)["data"].split(" ")[-1]
+            self._acm[src_ip] = access_code
+
             msg = f'Added amt://{self._mcast_src_ip}@{mcast_dst_ip}:{self._mcast_dst_port} to the Multicast Menu ' \
-                  f'(email={email}; description={description}).'
+                  f'(source={src_ip}). The access code is {access_code}.'
             print(msg)
+
         except Exception as e:
             errmsg += f'encountered a {type(e)} with message "{e}".'
             print(errmsg)
 
-    # TODO add functionality that evicts addresses from the FIB when they've been inactive for a while.
-
+    def _identify_inactive_streams(self):
+        """
+        Loop through the streams in the FIB and determine which have not been seen in a while.
+        """
+        try:
+            self._lock.acquire()
```

Comment:
add todo comment `# TODO figure out how to do more granular locking to avoid blocking translation thread for a long time`